server/src/main/java/org/opensearch/action/admin/indices/shrink/ResizeRequest.java

```diff
@@ -146,6 +146,29 @@ public ActionRequestValidationException validate() {
         if (maxShardSize != null && maxShardSize.getBytes() <= 0) {
             validationException = addValidationError("max_shard_size must be greater than 0", validationException);
         }
+        // Check target index's settings, if `index.blocks.read_only` is `true`, the target index's metadata writes will be disabled
```

Comment:
Yeah, it is, but the `read_only_allow_delete` block will be automatically released when the disk utilization falls below the watermark, did you mean we should also check the `read_only_allow_delete` block here?

Reply:
I was thinking specifically for case of `ResizeType.CLONE` (SHRINK/SPLIT I think are fine): it may happen than segments have to copied from source to target. When the node is low on disk space, resetting `read_only_allow_delete` would actually make things worst and easily lead to "out of free space".  May be we should be selective when offering the advice to reset `read_only_allow_delete` and for `CLONE` hint that additional disk space could be required?

Reply:
SHRINK, SPLIT and CLONE have same implementation, segments copy can happen in all of them if the filesystem does not support hard-linking. I think why we want to add more index blocks check for SHRINK,SPLIT and CLONE in this PR is that if user sets index setting `index.blocks.read_only` or `index.blocks.metadata` to `true` for the target index explicitly or inherit the index block from the source index, then the newly created target index's shards cannot recover successfully because of the block, so we want to prevent user from setting the block explicitly and they can reset the inherited block from the source index. Besides, `index.blocks.read_only_allow_delete` can also disable metadata writes for the newly created index, when the source index does have this block, as you said, if user reset this block to `null` for the newly created index, may lead to `out of free space`, so in my opinion, we don't check the `index.blocks.read_only_allow_delete` setting, if the source index has this block, the block will be copied to the newly created index, then the new shards in the newly created index will not recover successfully because of this block, it will prevent the disk from being out of usage, does this make sense?

Reply:
Sure, the way we deal with `index.blocks.read_only_allow_delete` does make sense,  I am concerned (may be too much) that we are offering potentially harmful the advice to user, do you think we could it make sense to remove this part ` please set \"index.blocks.metadata=false\" or \"index.blocks.metadata=null\"`?

```
 "target index ["
                    + targetIndexRequest.index()
                    + "] will be blocked by [index.blocks.read_only=true], this will disable metadata 
writes and cause the shards to be unassigned"
```

Reply:
Yeah, it makes sense, I'll change the error message.